ogr/ogrsf_frmts/sqlite/ogrsqlitelayer.cpp

```diff
@@ -188,6 +189,10 @@ static int OGRIsBinaryGeomCol(sqlite3_stmt *hStmt, int iCol,
     {
         eGeomFormat = OSGF_FGF;
     }
+    if (peGeometryType)
+    {
+        *peGeometryType = poGeometry ? poGeometry->getGeometryType() : wkbNone;
```

Comment:
Inferring the whole layer geometry type from the first geometry could be unreliable. I don't think we want to do this and the limitation can be removed, without scanning the whole layer, which would not scale well. So I think we must live with the limitation